pages/residents/[id]/shareable.tsx

```diff
@@ -0,0 +1,127 @@
+import CaseNoteGrid from 'components/ResidentPage/CaseNoteGrid';
+import { getResident } from 'lib/residents';
+import { GetServerSideProps } from 'next';
+import { Resident } from 'types';
+import { useCases } from 'utils/api/cases';
+import { isAuthorised } from 'utils/auth';
+import useWorkflows from 'hooks/useWorkflows';
+import { prettyResidentName } from 'lib/formatters';
+import Head from 'next/head';
+import s from '../../../stylesheets/Shareable.module.scss';
+import WorkflowTree from 'components/ResidentPage/WorkflowTree';
+import { useRelationships } from 'utils/api/relationships';
+import RelationshipsTable from 'components/ResidentPage/RelationshipsTable';
+
+interface Props {
+  resident: Resident;
+}
+
+const ResidentPage = ({ resident }: Props): React.ReactElement => {
```

Comment:
I'm not seeing any new tests to cover this page. Please add.

Reply:
there's no conventions for testing next.js pages in this repo and the equivalents in the other repo are too noodley for me to follow.

would probzbly look something like:

```
// pagesTest/residents/[id]/shareable.spec.tsx

import { fireEvent, render, screen } from '@testing-library/react';
import { mockedResident } from 'factories/residents';
import ShareableResidentPage from 'pages/residents/[id]/shareable';

global.window.print = jest.fn();

describe('shareable version', () => {
  it("renders resident's pretty name and page landmarks", () => {
    render(<ShareableResidentPage resident={mockedResident} />);
    expect(screen.getByText('Foo Bar'));
  });

  it('can trigger printing', () => {
    render(<ShareableResidentPage resident={mockedResident} />);
    fireEvent.click(screen.getByText('Print or save to PDF'));
    expect(window.print).toBeCalled();
  });

  it('shows resident details', () => {
    render(<ShareableResidentPage resident={mockedResident} />);
    expect(screen.getByText("Details"))
    expect(screen.getAllByRole("definition").length).toBe(???);
    // etc...
  });

  it('shows case notes', () => {
    render(<ShareableResidentPage resident={mockedResident} />);
    expect(screen.getByText("Case notes"))
    // etc...
  });

  it('shows workflows', () => {
    render(<ShareableResidentPage resident={mockedResident} />);
    expect(screen.getByText("Workflows"))
    // etc...
  });

  it('shows relationships', () => {
    render(<ShareableResidentPage resident={mockedResident} />);
    expect(screen.getByText("Relationships"))
    // etc...
  });

});

```

Reply:
Doesn't have to be a unit test, can be a cypress test instead. Looking over it there isn't great coverage of all pages, but we can't be introducing new stuff without at least having a baseline set of tests for them. I'd suggest adding a cypress test to cover the rendering of this page. Also wondering what this page does when the resident is restricted.